Source/Widgets/InfoBox/InfoBox.js

```diff
@@ -199,6 +172,125 @@ click: function () { closeClicked.raiseEvent(this); }');
 
         return destroyObject(this);
     };
+    /**
+     * Removes the frame if exists.
+     */
+    InfoBox.prototype.removeFrame = function() {
+        // cache the frame
+        var frame = this._frame;
+        // if no frame exists, do nothing
+        if (!(frame instanceof HTMLElement)) {
+            return;
+        }
+
+        // if we subscribed, we need to remove subscription
+        if (defined(this._descriptionSubscription)) {
+            this._descriptionSubscription.dispose();
+        }
+        frame.parentNode.removeChild(frame);
+        this._frame = null;
+    };
 
+    /**
+     * Creates a new frame according to allowScripts.
+     */
+    InfoBox.prototype.createFrame = function() {
+        this.removeFrame(); //make sure to remove existing frame if any
+        var frame = document.createElement('iframe');
+        frame.className = 'cesium-infoBox-iframe';
+        frame.setAttribute('sandbox', this.allowScripts ?
+                                      'allow-same-origin allow-popups allow-forms allow-scripts allow-modals' :
+                                      'allow-same-origin allow-popups allow-forms'); //allow-pointer-lock allow-scripts allow-top-navigation
+        frame.setAttribute('data-bind', 'style : { maxHeight : maxHeightOffset(40) }');
+        frame.setAttribute('allowfullscreen', true);
+        this._element.appendChild(frame);
+        this._frame = frame;
+
+        var that = this;
+        //We can't actually add anything into the frame until the load event is fired
+        frame.addEventListener('load', function() {
+            var frameDocument = frame.contentDocument;
+
+            //We inject default css into the content iframe,
+            //end users can remove it or add their own via the exposed frame property.
+            var cssLink = frameDocument.createElement('link');
```

Comment:
`frameDocument` throws `undefined` twice on an initial page load in IE11.  It doesn't appear to cause any problems, but a `defined` check may be needed here.

Reply:
I can't duplicate this.  Maybe it was a bug in a certain version of IE11 and it is now fixed?

Reply:
I can reproduce this on my machine (fully patched IE on Windows 10).  I can try and look into this for you @YonatanKra, but it may have to wait until the weekend.  Thanks again for the PR.

Reply:
I poked this a bit more, and as far as I can tell it's an IE11 bug.  I ended up sticking some loggers in the load event to watch the sandbox attribute, and the error throws only when the `DOMSettableTokenList` doesn't include `allow-scripts` as a value, and the `load` event fires multiple time in IE.

Logging out `contentDocument` and the `sandbox` attributes is really weird in IE and inconsistent.  The sandbox `DOMSettableTokenList` when logged, will show a `value` that includes `allow-scripts`, as well as a length of 4, but the logged object doesn't include `allow-scripts` in the pseudo array.  It's almost as if IE is firing a bad load event without scripts, then fires another load with scripts working.

A check for `null` to manage this behavior for IE might be the best fix.

Reply:
I still cannot reproduce this issue.  If you can add a fix quickly or tell me how to reproduce (I'm loading cesium with this feature both on IE11 and Edge - but do not see this error) we can resolve this issue.